tests/_base.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+from sklearn import exceptions
+from sklearn.utils.validation import check_X_y
+import unittest
+import time
+
+
+class PCovRTestBase(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.data = np.load('./tests/CSD-test.npz')
+        self.X = self.data["X"]
+        self.Y = self.data["Y"]
+        self.K = None
+        self.model = None
+
+        self.error_tol = 1E-3
+        self.rounding = -int(round(np.log10(self.error_tol)))
+
+        n_mixing = 11
+        self.lr_errors = np.nan * np.zeros(n_mixing)
+        self.pca_errors = np.nan * np.zeros(n_mixing)
+        self.alphas = np.linspace(0, 1, n_mixing)
+
+    def setUp(self):
+        self.startTime = time.time()
+
+    def tearDown(self):
+        t = time.time() - self.startTime
+        print('%s: %.3f' % (self.id(), t))
+
+    def rel_error(self, A, B):
+        return np.linalg.norm(A-B)**2.0 / np.linalg.norm(A)**2.0
+
+    def run_pcovr(self, mixing):
+        pcovr = self.model(mixing=mixing,
+                           n_components=2,
+                           tol=1E-12)
+        if(self.K is None):
+            pcovr.fit(self.X, self.Y)
+            Yp = pcovr.predict(self.X)
+            T = pcovr.transform(self.X)
+        else:
+            pcovr.fit(self.X, self.Y, K=self.K)
+            Yp = pcovr.predict(self.X, K=self.K)
+            T = pcovr.transform(self.X, K=self.K)
+        Xr = pcovr.inverse_transform(T)
+        return Yp, T, Xr
+
+    def add_errors(self):
+        for i, mixing in enumerate(self.alphas):
+            if(np.isnan(self.lr_errors[i]) or np.isnan(self.pca_errors[i])):
+                Yp, _, Xr = self.run_pcovr(mixing=mixing)
+                self.lr_errors[i] = self.rel_error(self.Y, Yp)
+                self.pca_errors[i] = self.rel_error(self.X, Xr)
+
+    # Basic Test of model LR Errors, that None return np.nan
+
+    def test_lr_errors(self):
+        self.add_errors()
```

Comment:
After this function, all lr_errors become not None, or I didn't understand something?

Reply:
The point is to run the model once per alpha and then use the results in multiple tests.